config/ompi_check_ucc.m4

```diff
@@ -0,0 +1,74 @@
+dnl -*- shell-script -*-
+dnl
+dnl Copyright (c) 2021      Mellanox Technologies. All rights reserved.
+dnl Copyright (c) 2013-2021 Cisco Systems, Inc.  All rights reserved.
+dnl Copyright (c) 2015      Research Organization for Information Science
+dnl                         and Technology (RIST). All rights reserved.
+dnl $COPYRIGHT$
+dnl
+dnl Additional copyrights may follow
+dnl
+dnl $HEADER$
+dnl
+
+# OMPI_CHECK_UCC(prefix, [action-if-found], [action-if-not-found])
+# --------------------------------------------------------
+# check if ucc support can be found.  sets prefix_{CPPFLAGS,
+# LDFLAGS, LIBS} as needed and runs action-if-found if there is
+# support, otherwise executes action-if-not-found
+AC_DEFUN([OMPI_CHECK_UCC],[
+    OPAL_VAR_SCOPE_PUSH([ompi_check_ucc_dir ompi_check_ucc_libs ompi_check_ucc_happy CPPFLAGS_save LDFLAGS_save LIBS_save])
+
+    AC_ARG_WITH([ucc],
+                [AS_HELP_STRING([--with-ucc(=DIR)],
+                                [Build UCC (Unified Collective Communication)])])
+
+    AS_IF([test "$with_ucc" != "no"],
+          [ompi_check_ucc_libs=ucc
+           AS_IF([test ! -z "$with_ucc" && test "$with_ucc" != "yes"],
+                 [ompi_check_ucc_dir=$with_ucc])
+
+           CPPFLAGS_save=$CPPFLAGS
+           LDFLAGS_save=$LDFLAGS
+           LIBS_save=$LIBS
+
+           OPAL_LOG_MSG([$1_CPPFLAGS : $$1_CPPFLAGS], 1)
+           OPAL_LOG_MSG([$1_LDFLAGS  : $$1_LDFLAGS], 1)
+           OPAL_LOG_MSG([$1_LIBS     : $$1_LIBS], 1)
+
+           OPAL_CHECK_PACKAGE([$1],
+                              [ucc/api/ucc.h],
+                              [$ompi_check_ucc_libs],
```

Comment:
Just curious: why is this a variable?  There's only one possible value it can be (`ucc`, set on line 27).